### PR TITLE
Update of the mesos.md getting started guide

### LINKED ISSUE #8957: Help needed to update the examples in getting-started-guides/mesos.md
Hi @ConnorDoyle @jdef  @guenter @nak3, 

We are updating docs to v1beta3. v1beta1 and v1beta2 will be abandoned soon. For the examples in getting-started-guides, we tend to use hyperlinks to point to examples instead of letting them have their own individual examples (see #5799). 

However, in the mesos guide, the examples contain mesos specific setup and knowledge, so I can't simply remove the texts and point to kubernetes/examples/guestbook, which is updated to v1beta3. My suggestion is that we update the guestbook example in kubernetes-mesos, and use a hyperlink in mesos getting-started-guide to replace the example texts. Could you help to make the changes? Thanks.

### PR DESCRIPTION
**Pull request by jdef**
Revise walkthrough to work with the newly upstreamed k8s-mesos code and modern v1 k8s.
- Addressed review comments
- update PATH rather than cd-ing and creating a conf file in a build output tree
- simplify instructions; we do not support the stock guestbook example in these instructions because it relies upon kube-dns and we do not set that up here
- moved note above 'get services' command
- link to dir/ not README.md
- address review feedback
- cat to create the mesos-cloud.conf contents
- remove unwanted (cloud_conf) underscore
- Update mesos.md
- removed http:// from MESOS_MASTER def
- MESOS_MASTER docs were wrong
- add missing zk port

xref 
- https://github.com/mesosphere/kubernetes-mesos/issues/295
- https://github.com/mesosphere/kubernetes-mesos/issues/209
- fixes #8957

/cc @davidopp @erictune @bgrant0607 @guenter @joerg84 @sttts

### CODE REVIEW

**Comment by jdef** [comment]
I've signed it, along with @sttts who is also doing work for Mesosphere.

**Comment by jdef** [comment]
xref https://github.com/GoogleCloudPlatform/kubernetes/pull/9853

File: docs/getting-started-guides/mesos.md [hunk 1]
```diff
@@ -5,20 +5,23 @@
 Mesos allows dynamic sharing of cluster resources between Kubernetes and other first-class Mesos frameworks such as [Hadoop][1], [Spark][2], and [Chronos][3].
 Mesos also ensures applications from different frameworks running on your cluster are isolated and that resources are allocated fairly.
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: maybe "fairly among them" to be a little clearer?

File: docs/getting-started-guides/mesos.md [hunk 2]
```diff
@@ -5,20 +5,23 @@
...
 
-Running Kubernetes on Mesos allows you to easily move Kubernetes workloads from one cloud provider to another to your own physical datacenter.
+Running Kubernetes on Mesos allows you to easily move Kubernetes workloads from one cloud provider to another or to your own physical datacenter.
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: I think this sentence in isolation is not useful unless there are instructions somewhere on how to do it (and it's actually easier than doing the same thing on native Kubernetes).

File: docs/getting-started-guides/mesos.md [hunk 3]
```diff
@@ -5,20 +5,23 @@
...
 
 This tutorial will walk you through setting up Kubernetes on a Mesos cluster.
 It provides a step by step walk through of adding Kubernetes to a Mesos cluster and running the classic GuestBook demo application.
-The walkthrough presented here is based on the v0.4.x series of the Kubernetes-Mesos project, which itself is based on Kubernetes v0.11.0.
+The walkthrough presented here is based on the v0.6.x series of the Kubernetes-Mesos project, which itself is based on Kubernetes v1.0.
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: How does a user know what version of the Kubernetes-Mesos project they are using? I think that for this sentence to be useful, you need to tell the user how they can check what Kubernetes-Mesos (and Kubernetes) version they're using.

File: docs/getting-started-guides/mesos.md [hunk 4]
```diff
@@ -5,20 +5,23 @@
...
 
-**NOTE:** There are [known issues with the current implementation][11].
-Please [file an issue against the kubernetes-mesos project][12] if you have problems completing the steps below.
+**NOTE:** There are [known issues with the current implementation][7].
+Please [file an issue against the kubernetes-mesos project][8] if you have problems completing the steps below.
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: can you add to that list the fact that you don't run the addons for monitoring, logging, and DNS? (obviously fine to do in a separate PR)

File: docs/getting-started-guides/mesos.md [hunk 5]
```diff
@@ -5,20 +5,23 @@
...
 
 ### Prerequisites
 
-* Understanding of [Apache Mesos][10]
+* Understanding of [Apache Mesos][6]
 * Mesos cluster on [Google Compute Engine][5]
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: Are there instructions on how to set this up, that you could link to from here? I think you used to link to google.mesosphere.com but I guess that doesn't set it up for you anymore.

File: docs/getting-started-guides/mesos.md [hunk 6]
```diff
@@ -5,20 +5,23 @@
...
-* A VPN connection to the cluster.
+* A VPN connection to the cluster
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: Any instructions for this?

File: docs/getting-started-guides/mesos.md [hunk 7]
```diff
@@ -42,9 +45,9 @@ The internal IP address of the master may be obtained via `hostname -i`.
 
 ```bash
 $ export servicehost=$(hostname -i)
-$ export mesos_master=${servicehost}:5050
 $ export KUBERNETES_MASTER=http://${servicehost}:8888
 ```
+
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: comment on 'sudo docker run' line a few lines below: this will almost always work, but the etcd version that we have actually qualified Kubernetes against can be found in "cluster/images/etcd/Makefile" which may not be the latest etcd. Probably not worth mentioning though.

File: docs/getting-started-guides/mesos.md [hunk 8]
```diff
@@ -133,175 +141,58 @@ Go to the Frameworks tab, and look for an active framework named "Kubernetes".
...
-},
-"labels": {
-  "name": "foo"
-} }
+$ cat <<EOPOD >nginx.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: to make this even simpler, you can drop this line and the next one

File: docs/getting-started-guides/mesos.md (hunk 1 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: fixed.

File: docs/getting-started-guides/mesos.md (hunk 2 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: Mesos has less requirements regards the network setup. With the k8sm endpoint controller this also applies to Kubernetes. In the follow-up commit I have expressed this more clearly.

File: docs/getting-started-guides/mesos.md (hunk 3 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: As this is more a matter of packaging k8sm and self-rerences to versions of the same repo do not make sense, I removed this.

File: docs/getting-started-guides/mesos.md (hunk 4 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: DNS in fact is working fine on-top of k8sm. Added monitoring and logging as not available yet.

File: docs/getting-started-guides/mesos.md (hunk 5 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: linked to instructions.

File: docs/getting-started-guides/mesos.md (hunk 6 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: linked to instructions.

File: docs/getting-started-guides/mesos.md (hunk 8 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: done

File: docs/getting-started-guides/mesos.md (hunk 7 above)
Comment (sttts) [inline docs/getting-started-guides/mesos.md]: The docker hub version is not maintained. With the follow-up commit we switch to the quai.io container with the same version as in cluster/images/etcd/Makefile (v2.0.9).

**Comment by davidopp** [comment]
I noticed you added some commits; let me know when it is ready to re-review.

**Comment by sttts** [comment]
@davidopp it's ready for your review

File: docs/getting-started-guides/mesos.md [hunk 9]
```diff
@@ -3,22 +3,25 @@
...
 
 ### Prerequisites
 
-* Understanding of [Apache Mesos][10]
-* Mesos cluster on [Google Compute Engine][5]
-* A VPN connection to the cluster.
+* Understanding of [Apache Mesos][6]
+* A running [Mesos cluster on Google Compute Engine][5]
+* A [VPN connection][10] to the cluster
+* On the master node:
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: maybe say "On the Mesos master node" just to be crystal clear.

File: docs/getting-started-guides/mesos.md [hunk 10]
```diff
@@ -42,65 +45,72 @@ The internal IP address of the master may be obtained via `hostname -i`.
...
+```
+Create a cloud config file `mesos-cloud.conf` in the current directory with the following contents:
+```bash
+$ cat <<EOF >mesos-cloud.conf
+[mesos-cloud]
+        mesos-master        = ${MESOS_MASTER}
+EOF
+```
+
+Now start the kubernetes-mesos API server, controller manager, and scheduler on a Mesos master node:
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: should this be "the Mesos master node" ? I know you can do HA masters so in that situation there can be multiple, but if the user follows the default mesos setup instructions do they get that?

**Comment by davidopp** [comment]
looks good; just had a few nits (see above). can you please address those and then squash the commits and then it'll be ready to merge. Thanks!

**Comment by sttts** [comment]
@davidopp clarified the use of "master" everywhere, and squashed. Ready for merge.

**Comment by davidopp** [comment]
Can you really run the commands in "prerequisites" and "Deploy Kubernetes-Mesos" on any node in the Mesos cluster? Is so, please mention that explicitly as I think people might expect otherwise. If not, please clarify it anyway. :)

Other than that, looks good.

**Comment by sttts** [comment]
@davidopp clarified which machine k8sm can be deployed on. And squashed again.

**Comment by davidopp** [comment]
LGTM

Looks great. Thanks!

This is documentation-only so can be merged.

**Comment by davidopp** [comment]
BTW you can ignore the CLA error, it's a bug.
